pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -747,3 +756,45 @@ func addressSubset(addresses []v1.EndpointAddress, maxNum int) []v1.EndpointAddr
 	}
 	return addresses[0:maxNum]
 }
+
+// managedByController returns true if the endpoint is or has been managed by this endpoints controller
+func managedByController(endpoint *v1.Endpoints) bool {
+	// skip leader election endpoints
+	if _, ok := endpoint.Annotations[resourcelock.LeaderElectionRecordAnnotationKey]; ok {
+		return false
+	}
+
+	// skip kubernetes.default that is the endpoint managed by the apiserver
+	if endpoint.Name == "kubernetes" && endpoint.Namespace == "default" {
+		return false
+	}
+
+	// ignore endpoints without SSA fields for backwards compatibility
+	if len(endpoint.ManagedFields) == 0 {
+		return false
+	}
+
+	// if the endpoint has been managed by this controller process it
+	for _, f := range endpoint.ManagedFields {
+		if f.Manager == controllerName {
+			return true
+		}
+	}
+	return false
```

Comment:
This might be safer:

```suggestion
	for _, f := range endpoint.ManagedFields {
		if f.Manager != controllerName {
			return false
		}
	}
	return true
```

Reply:
I want to solve the situation that an user changes an endpoint that was created by this controller, the endpoint still belongs to the controller and it should be processed again ... we had an issue about this that I can't find.
So, if the endpoint has at least one field with this controller as manager we process it ....

Reply:
I just realized that this is larger in scope than I initially realized... I think the only thing this should guard is whether or not we cleanup/delete a given Endpoints resource, not whether we modify it. 

With EndpointSlices we could choose not to modify an EndpointSlice resource and just create/update different EndpointSlices for the Service if necessary. That's not possible with Endpoints and the 1:1 name matching. The current approach in this PR risks Endpoints being taken over forever by user modification with no way for the Endpoints controller to recover.

Reply:
Hmm, maybe I'm not understanding, right now taken over endpoints is a fact, since the controller doesn't process update events it will not "recover" it if an user modifies it

Reply:
Currently the controller will continue to manage Endpoints for Services with a selector, even if the user modifies them. I think we need that to continue. The only real behavior change we need here is for the controller to clean up Endpoints it's managing after the Service goes away or loses its selector.

Reply:
> since the controller doesn't process update events it will not "recover" it if an user modifies it

I think that's somewhat orthogonal to this. It can't really be solved without a handler for Endpoint updates, and even then, it would take some safeguards like the EndpointSliceTracker to implement well.